features/networking/multus.feature

```diff
@@ -416,4 +416,248 @@ Feature: Multus-CNI related scenarios
+    the bridge interface named "mybridge" is deleted from the "<%= cb.nodes[1].name %>" node
+    the bridge interface named "mybridge.100" is deleted from the "<%= cb.nodes[1].name %>" node
+    """  
+    #making sure the pods on same node can ping while pods on diff nodes can't
+
+    When I execute on the "<%= cb.pod1 %>" pod:
+      | ping | -c1 | -W2 | <%= cb.pod2_net1_ip %> |
+    Then the step should succeed
+
+    When I execute on the "<%= cb.pod2 %>" pod:
+      | ping | -c1 | -W2 | <%= cb.pod1_net1_ip %> |
+    Then the step should succeed
+    
+    When I execute on the "<%= cb.pod1 %>" pod:
+      | ping | -c1 | -W2 | <%= cb.pod3_net1_ip %> |
+    Then the step should fail
+
+    When I execute on the "<%= cb.pod3 %>" pod:
+      | ping | -c1 | -W2 | <%= cb.pod1_net1_ip %> |
+    Then the step should fail
+    
+    When I execute on the "<%= cb.pod3 %>" pod:
+      | ping | -c1 | -W2 | <%= cb.pod2_net1_ip %> |
+    Then the step should fail
+    
+    When I execute on the "<%= cb.pod2 %>" pod:
+      | ping | -c1 | -W2 | <%= cb.pod3_net1_ip %> |
+    Then the step should fail
+    
+  # @author anusaxen@redhat.com
+  # @case_id OCP-24491
+  @admin
+  Scenario: Pods cannot communicate each other with different vlan tag
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    And I store all worker nodes to the :nodes clipboard
+    # Create the net-attach-def with vlan 100 via cluster admin
+    Given I have a project
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/bridge-host-local-vlan.yaml |
+      | n | <%= project.name %>                                                                                                                               |
+    Then the step should succeed 
+    
+    # Create the net-attach-def with vlan 200 via cluster admin
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/bridge-host-local-vlan-200.yaml |
+      | n | <%= project.name %>                                                                                                                                   |
+    Then the step should succeed 
+    #Labeing a worker node to make sure all future pods to be scheduled on this node only
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test=worker1            |
+    Then the step should succeed
+    
+    Given I register clean-up steps:
+    """
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test-                   |
+    Then the step should succeed
+    """ 
```

Comment:
we have a step ` /^label #{QUOTED} is added to the#{OPT_QUOTED} node$/`. It automatically registers clean-up so scenario will become a lot shorter.

Reply:
hmm, wasn't aware of that, Thanks WTAL

Reply:
afraid, this step "/^label #{QUOTED} is added to the#{OPT_QUOTED} node$/" is not getting cleanup post scenario as per testing

Reply:
Could you provide full run log where you observe label not being restored?

Reply:
Line 1457 here http://pastebin.test.redhat.com/820129. I confirmed after the test by manaully login into cluster that label test=worker1 was still there on the node

Reply:
@akostadinov @pruan-rht Any thoughts?

Reply:
The issue has nothing to do with the label step. I see at the end
```
      [23:29:43] INFO> === After Scenario: Pods cannot communicate each other with different vlan tag ===
      [23:29:43] INFO> Step: the bridge interface named "mybridge.200" is deleted from the "ip-10-0-131-252.ap-northeast-2.compute.internal" node
      [23:29:43] ERROR> Test Execution will finish prematurely.
      Undefined dynamic step: "the bridge interface named "mybridge.200" is deleted from the "ip-10-0-131-252.ap-northeast-2.compute.internal" node" (Cucumber::UndefinedDynamicStep)
      /home/anusaxen/git/verification-tests/lib/world.rb:507:in `block (3 levels) in to_step_procs'
      /home/anusaxen/git/verification-tests/lib/world.rb:429:in `block in after_scenario'
      /home/anusaxen/git/verification-tests/lib/world.rb:429:in `reverse_each'
      /home/anusaxen/git/verification-tests/lib/world.rb:429:in `after_scenario'
      /home/anusaxen/git/verification-tests/lib/manager.rb:35:in `clean_up'
      /home/anusaxen/git/verification-tests/features/support/env.rb:68:in `block (2 levels) in <top (required)>'
      /home/anusaxen/git/verification-tests/lib/base_helper.rb:131:in `capture_error'
      /home/anusaxen/git/verification-tests/features/support/env.rb:66:in `After'
 ```

This means the clean-up step does not exist. So clean-up process ends before the label clean-up kicks in. Also the whole test run finishes pre-maturely.

Reply:
@akostadinov Yep, my bad. SOmething bad might have happened on that cluster as lot of tests were tried on it. I optimized the label steps across whole PR and ran tests 3 times. Test ends maturely and seems good to me now. 1650 and 4452 explains that - http://pastebin.test.redhat.com/822512.